Robus/src/timestamp.c

```diff
@@ -8,6 +8,75 @@
+ * the same moment.
+ *
+ * When you need to read the timestamp associated with an event, you can use the Timestamp_GetToken()
+ * function. This function will search the token linked to the event in the list and return the
+ * timestamp saved in it.
+ *
+
+                                              Event_A         Event_B
+
+                                                 │               │
+                                                 ▼               ▼
+                                            ───────────────────────────────────────────────►  Time
+                                                 |               |
+                                                 │               │
+               Token list                        │               │
+                                                 │               │
+               ┌───────┐                         │               │
+    Timestamp  │ Token │ ◄───────────────────────┘               │
+               ├───────┤                                         │
+    Timestamp  │ Token │ ◄───────────────────────────────────────┘
+               ├───────┤
+    Timestamp  │ Token │          Timestamp_Tag(token, Event_B)
+               ├───────┤
+    Timestamp  │ Token │
+               └───────┘
+                   .
+                   .
+                   .
+
+               ┌───────┐
+    Timestamp  │ Token │ ────────────────► Timestamp_GetToken(Event_B)
+               ├───────┤
+    Timestamp  │ Token │
+               ├───────┤
+    Timestamp  │ Token │
+               └───────┘
+
+ *
+ * You can also send a message with a data and it's associated timestamp, luos can handle this by slighly modifiyng its
+ * protocol. To encode a message with the timestamp protocol, use Timestamp_EncodeMsg() function, it will change the
+ * message is encoded and save the timestamp in it.
+ * An example is shown beneath: the sended data is a color, the the command saved in the header is COLOR_TYPE. What
+ * Timestamp_EncodeMsg() does is copying the COLOR_TYPE in a dedicated section after data payload and replace it in the
+ * header by a TIMESTAMP command. The timestamp value is placed at the end of the message, after data and data_cmd sections.
+ *
+                                          Timestamp            Color value             Color type
+ ┌─────────┬─────────┬─────────┬─────────┬─────────┬─────────┬───────────────────────┬─────────────┬────────────────┐
+ │Protocol │ Target  │  Mode   │ Source  │   Cmd   │  Size   │         Data          │  Data cmd   │   Timestamp    │
+ └─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴───────────────────────┴─────────────┴────────────────┘
+                                              │                                             ▲
+                                              │                                             │
+                                              └─────────────────────────────────────────────┘
```

Comment:
The protocol section of messages has been specifically designed to deal with this kind of slight protocol modifications we would want to make in the future.
You should definitely consider reserving a bit for it in the Protocol section avoiding a custom Timestamp command.
This could allow you to keep the information that this message is timestamped directly in the message and avoid any Cmd modifications.
Timestamp at the end sounds like the best option.